infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -517,18 +537,35 @@ public void updateTopology(LogReplicationClusterInfo.TopologyConfigurationMsg to
      * msg needs to send out.
      */
     @Override
-    public void prepareSiteRoleChange() {
-        replicationManager.prepareSiteRoleChange();
+    public void prepareClusterRoleChange() {
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && replicationManager != null) {
+            replicationManager.prepareClusterRoleChange();
+        } else {
+            log.warn("Illegal prepareSiteRoleChange when cluster{} with role {}",
```

Comment:
prepareSiteRoleChange -> prepareClusterRoleChange in the log comment.

Reply:
Can't there be a role change from standby to active?

Reply:
Typo fixed.

Yeah, it can be a role change from standby to active. I think it depends on how the cluster manager uses it...

I don't want to touch prepareClusterRoleChange() and queryReplicationStatus()'s functionality in this PR. I just added some logs and make it a little robust. It is better to have a new PR about those two functions.

Reply:
ok, let's leave it for another PR, but we should revisit this.